### PR TITLE
Avoid suggestion box calculation on firs input focus

### PR DESCRIPTION
**Pull request by perunt**
### Fixed Issues

$ https://github.com/Expensify/App/issues/47661
PROPOSAL: https://github.com/Expensify/App/issues/47661#issuecomment-2360841690

### Tests

You can also use all test cases described in this PR https://github.com/Expensify/App/pull/44113

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Launch app
2. Tap fab -- start chat -- room
3. Create a room
4. Enter @[jaihanumanblig@gmail.com](mailto:jaihanumanblig@gmail.com)
5. Send the message
6. In a whisper message, tap invite
7. Long press the message sent in step 4, and select edit comment
8. No suggestion box shouldn't appear when the input is focused first time

1. Open any chat 
2. Open the emoji picker menu
3. Select a skin tone (other than the default yellow)
4. Verify that the hand emoji has the proper selected color

1. Close and reopen the app
2. Open any chat 
3. Open the emoji picker menu
4. Verify that the hand emoji still has the previously selected color

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/c30eb509-b7aa-410e-9e53-9b92b71c0029

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/8735fef5-44d7-4125-9788-0caf76611436

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/7b7817e7-f16a-4370-9804-7159ff0d741d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/6b2d4fef-ec00-43e7-ac69-07bace6ecbaf

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/4dadaf5b-9dd3-4499-8dce-6fd1ac47fd82

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/f55f4573-deab-488b-be46-a09ecfc854a1

</details>

### CODE REVIEW

**Comment by perunt** [comment]
The error says `withOnyx is deprecated. Use useOnyx instead of withOnyx whenever possible`' This `withOnyx` was in a file I edited previously. I assume it would be replaced in all occurrences by a single PR

**Review by mkhutornyi — COMMENTED** [review]
Please add platform screenshots

File: src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ function SuggestionEmoji(
 ) {
     const [suggestionValues, setSuggestionValues] = useState(defaultSuggestionsValues);
     const suggestionValuesRef = useRef(suggestionValues);
+    const isInitialFocusRef = useRef(true);
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx:64]: Let's do this in a separate PR as it's not related here. This kind of replacement should be made as a migration and be applied across the whole app

[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx:64]: I suggest we address this in a separate pull request, as it's not directly related to the current changes. This type of replacement would be best implemented as a migration and applied consistently across the entire app.

[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx:64]: I can raise PR with that

**Comment by perunt** [comment]
> Please add platform screenshots

ahh, sorry about that, added

**Review by mkhutornyi — APPROVED** [review]

**Review by mkhutornyi — APPROVED** [review]

**Review by Gonals — APPROVED** [review]

**Comment by rlinoz** [comment]
We reverted this PR to fix https://github.com/Expensify/App/issues/50283

cc: @perunt @mkhutornyi

**Pull request merged**